### ISSUE TITLE
Branch Cleanup

### ISSUE THREAD

**Issue by MatthewChristoff** [SCOREC/pumi-pic#62]
The repository is probably due for some branch cleanup. I've noticed that we have quite a few inactive/old branches lying around that should be dealt with (merged/deleted).

First off: `cabmBuild` and `csrMigrate` should be merged into master when possible. We've already started on this but we're waiting on a few things, [request here](https://github.com/SCOREC/pumi-pic/pull/60). Note that when we merge in `csrMigrate`, we should attempt to make it easier on us by splitting it into `csrMigrate` and `cabmSwap` branches (my mistake). There's quite a lot of edits on these files, and I expect a few conflicts.

In terms of inactive branches:
`xgcp_mesh` hasn't been edited since last January, and has no commits ahead of `master`
`testShuffle` hasn't been edited since last January, and has 7 commits ahead of `master`
`siampp20_pumipicRuns` hasn't been edited since last February, and has 1 commit ahead of `master`
`pscopy` hasn't been edited since November, and has 4 commits ahead of `master`
`optimizeMigrate` hasn't been edited since last July, and has no commits ahead of `master`
`gd/timing` was edited 24 days ago, but has no commits ahead of `master`
`120kC16runs` hasn't been edited since two Decembers ago, and has 4 commits ahead of `master`
`csr` has been edited recently but should probably be merged into master and have another branch made

Besides those listed, there are quite a few other branches which should probably be taken a look at whether they need to be merged or not. Most of them have fairly few commits and are fairly far behind `master`.

**Comment by diamog** [comment]
The following branches have been deleted:
`xgcp_mesh`
`optimizeMigrate`
`gd/timing` The final set of features planned for this branch has been finished and merged into master

A short description for each of the remaining branches in question:
`testShuffle` - Attempts at measuring the performance gain of shuffling. Inconclusive and quite outdated now.
`siampp20_pumipicRuns` - Backup of changes for runs used for a presentation.
`pscopy` - Holds a feature that still needs testing on Summit to see if it's worthwhile. It was less performant on AiMOS.
`120kC16runs` - Assuming this is a backup of changes for runs used in a paper/presentation?
`csr` - Looks like this has some changes for GITRm, thus it shouldn't be merged into master at this time.

I think we should be fine to delete the `testShuffle` branch. If we wish to do performance tests on shuffling again, then it may be easier to start from scratch than outdated inconclusive test code.

@cwsmith How do you think we should handle the branches which hold backups for performance runs?

Additionally, we have three GITRm branches: `gitrm`, `fun3dmesh`, and `gitrm_memleak_fix`. I'll talk to the GITRm developers to see if I can reduce that to just one until I have the will to work on merging into master.

**Comment by diamog** [comment]
Thanks for testing that out. Sure you can do the same.

**Comment by cwsmith** [comment]
`siampp20_pumipicRuns` archive: https://zenodo.org/record/4648098

`siampp20_pumipicRuns` has been deleted.

**Comment by diamog** [comment]
Confirmed with Dhyan that we only need the `fun3dmesh` branch. The other two GITRm branches have been deleted

**Comment by MatthewChristoff** [comment]
@cwsmith Oh, I forgot about `csr_flat_for`. Did you say that was a feature that didn't work out? Is that a branch we should hold on to, or get rid of?

**Comment by cwsmith** [comment]
If @stevespreiz agrees, I think we can delete `csr_flat_for`.  I'm not sure about the others offhand.

**Comment by cwsmith** [comment]
<img src="https://user-images.githubusercontent.com/1024438/113156356-e804ff00-9207-11eb-884d-7d67246801f6.jpg" height="300" width="300"/>

**Comment by stevespreiz** [comment]
I also deleted some branches I don't use anymore:
`gpu_dist`
`gpu_dist_reorder`
`csr_histogram`
`memory_comparison`
`ps_profile`
